### PR TITLE
Adding UpdatedVoxelLayer

### PR DESCRIPTION
**Pull request by AlexReimann**
Makes VoxelLayer keep track of what costmap cells are updated
Adds ExtendedVoxelLayer which adds the option to update with differential overwrite instead of bounding box
Adds publishing of which cells are cleared by the ExtendedVoxelLayer
Added type info output about costmap plugins to ROS_INFO

### CODE REVIEW

**Comment by AlexReimann** [comment]
@bit-pirate 

updated, merge ready~

Please check

File: costmap_2d/plugins/voxel_layer.cpp [hunk 1]
```diff
@@ -354,6 +356,43 @@ void VoxelLayer::raytraceFreespace(const Observation& clearing_observation, doub
   }
 }
 
+void VoxelLayer::updateUpdateCells(boost::shared_ptr<bool[]> updated_columns, unsigned int updated_area_width, unsigned int offset_x, unsigned int offset_y)
```
Comment (bit-pirate) [inline costmap_2d/plugins/voxel_layer.cpp:359]: maybe you can come up with a better name for this method :-)

File: costmap_2d/include/costmap_2d/extended_voxel_layer.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/*
```
Comment (bit-pirate) [inline costmap_2d/include/costmap_2d/extended_voxel_layer.h:1]: remove these kind of comments

**Comment by AlexReimann** [comment]
Merged to different branch:
https://github.com/yujinrobot/navigation/pull/10

**Pull request closed without merging**